### PR TITLE
[model, rollout, data, reward, cfg] feat: add Hindi Qwen3-TTS SFT-LoRA GRPO recipe

### PR DESCRIPTION
**Pull request by dongbo910220**
### What does this PR do?

Adds a reproducible Hindi SFT-LoRA GRPO recipe for Qwen3-TTS.

This PR adds a verified conversion for the published Hindi MLX SFT adapter, deterministic IndicVoices-R Hindi train preparation, a standalone Whisper CER HTTP scorer, a LoRA training launcher, focused CPU tests, and documentation.

Duplicate-work checks found no open PR for this Hindi SFT-LoRA GRPO recipe ([search](https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+is%3Aopen+Qwen3-TTS+Hindi+GRPO)).

### Test

```bash
.venv/bin/python -m pytest -q \
  tests/pipelines/test_qwen3_tts_hindi_recipe_on_cpu.py \
  tests/pipelines/test_qwen3_tts_on_cpu.py \
  tests/utils/reward_score/test_whisper_cer_scorer_on_cpu.py
```

Result: `all passed`.

A 400-update run evaluated the same complete 100-prompt validation set at step 0 and every 20 updates.

Qwen3-TTS Hindi GRPO reward and validation curve

Qwen3-TTS Hindi GRPO capped CER curve

Validation reward increased from `0.7992` at Base to `0.8313` at step 400 and peaked at `0.8422` at step 340. Capped CER decreased from `0.1908` to `0.1612` at step 400 and reached `0.1482` at step 380. 

### Design & Code Changes

- **Initial policy and data:** verifies and merges the published Hindi SFT adapter into a pinned non-quantized Base checkpoint, then builds disjoint 863/100 IndicVoices-R Hindi splits from a pinned revision.
- **Training recipe:** starts a fresh rank-8 LoRA from the merged SFT policy with `B=4`, `G=4`, LR `5e-6`, `k3` KL coefficient `0.08`, 400 updates, checkpoints every 20 updates, and the complete fixed 100-prompt validation set every 20 updates.

Merged checkpoint: [Qwen3-TTS-12Hz-0.6B-Hindi-SFT-Merged](https://huggingface.co/littleFishhhh/Qwen3-TTS-12Hz-0.6B-Hindi-SFT-Merged)

### CODE REVIEW

**Comment by dongbo910220** [comment]
@zhtmike Could you take a look at this when you have a chance? Thanks!

**Comment by zhtmike** [comment]
Sure!

**Comment by NancyFyong** [comment]
minor comments, can you share this merged lora base model to huggingface so that users can use it directly?

**Comment by dongbo910220** [comment]
@NancyFyong  Uploaded the merged checkpoint to [littleFishhhh/Qwen3-TTS-12Hz-0.6B-Hindi-SFT-Merged](https://huggingface.co/littleFishhhh/Qwen3-TTS-12Hz-0.6B-Hindi-SFT-Merged)

**Comment by NancyFyong** [comment]
LGTM

**Review by NancyFyong — APPROVED** [review]

File: examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py
Comment (zhtmike) [inline examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py:1]: let us move 
`examples/grpo_trainer/qwen3_tts/whisper_cer_scorer.py` -> `examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py`

File: examples/grpo_trainer/qwen3_tts/whisper_cer_scorer.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    prompt = payload.get("prompt")
+    if not isinstance(prompt, str) or not normalize_text(prompt):
+        raise ValueError("Audio scorer requires a non-empty prompt.")
+    metadata = payload.get("metadata")
+    if not isinstance(metadata, dict):
+        raise ValueError("metadata must be a JSON object.")
+    return waveform, sample_rate, prompt, metadata
+
+
+class WhisperCERScorer:
```
Comment (zhtmike) [inline examples/grpo_trainer/qwen3_tts/whisper_cer_scorer.py]: write a docstring of this Scorer, so other model can reuse it.

File: examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import json
+import re
+import threading
+import unicodedata
+from http import HTTPStatus
+from http.server import BaseHTTPRequestHandler, ThreadingHTTPServer
+from typing import Any
+
+import numpy as np
+
```
Comment (zhtmike) [inline examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py:31]: move the scorer (main APIs) at the top of the file and add `__all__` for it,
move the helper utilities at the bottom

File: tests/pipelines/test_qwen3_tts_package_on_cpu.py [hunk 3]
```diff
@@ -183,3 +184,4 @@ def test_qwen_tts_registers_and_runs_without_a_transformers_compatibility_layer(
     assert all(name.startswith(("talker.model.", "talker.codec_head.")) for name in trainable_names)
     assert any(not parameter.requires_grad for parameter in configured.parameters())
     assert configured.get_input_embeddings() is configured.talker.model.codec_embedding
+    configured.config.save_pretrained(tmp_path / "checkpoint_config")
```
Comment (zhtmike) [inline tests/pipelines/test_qwen3_tts_package_on_cpu.py]: need to clean the test things after test is finished

File: verl_omni/pipelines/qwen3_tts/talker_training_adapter.py
Comment (zhtmike) [inline verl_omni/pipelines/qwen3_tts/talker_training_adapter.py:1]: let us add a TODO for this,  e.g., use transformers version in future

File: examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py
Comment (dongbo910220) [reply examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py:1]: moved

File: examples/grpo_trainer/qwen3_tts/whisper_cer_scorer.py (hunk 1 above)
Comment (dongbo910220) [reply examples/grpo_trainer/qwen3_tts/whisper_cer_scorer.py]: added

File: examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py (hunk 2 above)
Comment (dongbo910220) [reply examples/grpo_trainer/qwen3_tts/scorer/whisper_cer.py:31]: Moved `WhisperCERScorer` above the helper utilities and added `__all__ = ["WhisperCERScorer"]`.

File: tests/pipelines/test_qwen3_tts_package_on_cpu.py (hunk 3 above)
Comment (dongbo910220) [reply tests/pipelines/test_qwen3_tts_package_on_cpu.py]: updated the test to use `TemporaryDirectory`

File: verl_omni/pipelines/qwen3_tts/talker_training_adapter.py
Comment (dongbo910220) [reply verl_omni/pipelines/qwen3_tts/talker_training_adapter.py:1]: added

**Review by zhtmike — APPROVED** [review]

**Comment by dongbo910220** [comment]
@zhtmike I’ve fixed the CPU test failure and pushed the update. Could you please re-add the `ready-for-ci` label ?

**Comment by zhtmike** [comment]
Nice work, All CI green.
